### ISSUE TITLE
Gatsby v2 - IE11 Support

### ISSUE THREAD

**Issue by lovenick** [gatsbyjs/gatsby#9034]
## Description

Gatsby / Babel transforms ES5 object properties to ES6 shorthand properties. 

When testing in IE11 with Gatsby and Firebase, the Firebase JS throws `Expected ':'`. Upon inspection, the minified source code includes:

`{..., apps:null,Promise,SDK_VERSION:"5.5.0", ...}` 

Obviously IE chokes on the shorthand property. It doesn't look like the Firebase module ships with shorthand properties, so it must occur in Gatsby transpiling. Plus, I was able to test the theory by manually changing the variable names in the Firebase module (from `{..., Promise: Promise, ...}` to `{..., Promise: _Promise, ...}`) which corrected the issue.

It appears the Babel presets don't affect node_modules, but still transpile the source. 

So is this a Gatsby config issue or Babel?

### Issue 1
1. use 'firebase/app' module in a Gatsby v2 project
2. build and test in IE11

### Expected result

It should render.

### Actual result

Blank screen with console errors.

### Environment

System:
    OS: macOS High Sierra 10.13.6
    CPU: x64 Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
    Shell: 5.3 - /bin/zsh

Binaries:
    Node: 8.11.1 - ~/.nvm/versions/node/v8.11.1/bin/node
    Yarn: 1.9.4 - /usr/local/bin/yarn
    npm: 6.4.1 - ~/.nvm/versions/node/v8.11.1/bin/npm
  
Browsers:
    Chrome: 69.0.3497.100
    Firefox: 62.0.2
    Safari: 12.0
  
npmPackages:
    gatsby: ^2.0.21 => 2.0.21
    gatsby-cli: ^2.4.3 => 2.4.3
    gatsby-plugin-favicon: ^3.1.4 => 3.1.4
    gatsby-plugin-sentry: ^0.1.0 => 0.1.0
    gatsby-plugin-styled-components: ^3.0.0-rc.5 => 3.0.0-rc.5
    gatsby-source-contentful: ^2.0.3 => 2.0.3

npmGlobalPackages:
    gatsby-cli: 2.4.3

**Comment by shanekenney** [comment]
I am seeing a similar issue, changing the terser ecma version to v5 in `webpack-utils.js` seems to fix this by preventing terser from using shorthand properties.
https://github.com/gatsbyjs/gatsby/blob/a42d0de654e70be1b25697848529e118e6bf92c6/packages/gatsby/src/utils/webpack-utils.js#L448

Is there any easy way to configure minification options?

**Comment by sheakelly** [comment]
@shanekenney I was able to solve this with the following gatsby-node.js changes

```
exports.onCreateWebpackConfig = ({
  stage,
  plugins,
  actions
}) => {
  switch (stage) {
    case 'build-javascript':
      actions.setWebpackConfig(
        optimization: {
          minimizer: [
            plugins.minifyJs({
              terserOptions: {
                ecma: 5 // Ensures minified js is IE11 compatible
              }
            }),
            plugins.minifyCss()
          ]
        }
      })
  }
}
```

This merges the updated optimization webpack config into the standard webpack.config.js used by gatsby

**Comment by kakadiadarpan** [comment]
Related to #8905 

Closed in favor of #8905
